app/users/create-user/createUserInteractor.js

```diff
@@ -0,0 +1,27 @@
+const ValidationError = require('../../errors/validationError');
+
+class CreateUserInteractor {
+    constructor ({presenter, validator, userRepository, userFactory, uuid}) {
+        this.presenter = presenter;
+        this.validator = validator;
+        this.userRepository = userRepository;
+        this.userFactory = userFactory;
+        this.uuid = uuid;
```

Comment:
This fancy generator details should be hidden behind fancy interface.
Maybe somewhat like `class IdGenerator { construtor({uuid}) {}  generate() {}, etc.}`